gemini/sample-apps/accelerating_product_innovation/app/pages_utils/utils_setup.py

```diff
@@ -0,0 +1,178 @@
+        st.session_state.previous_product_category = st.session_state.product_category
+        st.rerun()
+
+
+def get_session_state() -> dict:
+    """
+    Defines default values for session state
+
+    Returns:
+        session_state_defaults (dict): A dictionary of default key-value pairs
+        for session state
+    """
+    session_state_defaults: dict[str, Any] = {
+        "product_categories": utils_project.get_projects_list(),
+        "new_product_category_added": None,
+        "previous_product_category": None,
+        "text_edit_prompt": None,
+        "update_text_btn": None,
+        "uploaded_files": None,
+        "rag_search_term": None,
+        "rag_answers_gen": False,
+        "rag_answer": None,
+        "rag_answer_references": None,
+        "insights_suggestion": None,
+        "insights_placeholder": "",
+        "suggestion_first_time": 1,
+        "processed_data_list": [],
+        "query_vectors": [],
+        "dff": None,
+        "temp_suggestions": None,
+        "assorted_prod_title": None,
+        "assorted_prod_content": [],
+        "email_gen": False,
+        "create_product": False,
+        "modifying": False,
+        "custom_prompt": "",
+        "feature_suggestions": None,
+        "selected_titles": [],
+        "saved_titles": [],
+        "selected_prompt": None,
+        "product_gen_image": None,
+        "features_generated": False,
+        "generated_points": None,
+        "content_generated": False,
+        "product_content": None,
+        "image_to_edit": -1,
+        "generate_images": False,
+        "image_prompt": None,
+        "image_file_prefix": "uploaded_image",
+        "email_image": None,
+        "email_prompt": "High SPF",
+        "num_drafts": None,
+        "email_text": None,
+        "generated_image": None,
+        "mask_image": None,
+        "edit_suggestion": False,
+        "suggested_images": None,
+        "uploaded_img": False,
+        "start_editing": False,
+        "text_to_edit": None,
+        "content_edited": None,
+        "row": None,
+        "edited_content": None,
+        "col": None,
+        "generated_response": None,
+        "draft_elements": None,
+        "chosen_titles": [],
+        "buffer": None,
+        "save_edited_image": None,
+        "email_files": [],
+        "image_edit_col": None,
+        "image_edit_row": None,
+        "bg_editing": False,
+    }
+
+    return session_state_defaults
+
+
+def initialize_all_session_state():
+    """Initializes all the session states used in the app.
+
+    Args:
+        None
+
+    Returns:
+        None
+    """
+    session_state_defaults = get_session_state()
+    for key, value in session_state_defaults.items():
+        if key not in st.session_state:
+            st.session_state[key] = value
+
+    if "product_category" not in st.session_state:
+        st.session_state.product_category = st.session_state.product_categories[0]
+
+
+def reinitialize_session_states():
```

Comment:
Some of this is duplicated from other methods.

Reply:
merged with initialize_all_session_states.